### DISCUSSION TITLE
thread to shoulder bug

### DISCUSSION THREAD

**Discussion by tomvikse** [kwackers/electronic-leadscrew#3] [General]
Testing it out but there is a bug.
When is supose to show stop at shoulder is show nothing just RPM.. but when you stop the spinden is say SH 0..
Some times it jump back to normal(showing stop at shoulder) an back to SH 0 again..

**Comment by tomvikse** [comment]
https://photos.app.goo.gl/GYSws24Rh4fcKsPd6

https://photos.app.goo.gl/dCrMF4i4bcoSd2cQ9

**Reply by kwackers** [reply]
I can't quite get a handle on what you're showing me.
There was a bug where sometimes it didn't realise it had reached the shoulder - is that it?
If so try the latest, that should be fixed.

**Comment by tomvikse** [comment]
It say SH 0.   As you see in the last video..  return to start dont work, doing it manual it work .. and it stop at the set stop point

**Reply by kwackers** [reply]
The fix I just pushed up should fix that (I fixed it about a month ago but hadn't pushed up the fix).

**Reply by kwackers** [reply]
Did you try this? Did it fix it?

**Reply by tomvikse** [reply]
Test later

**Reply by tomvikse** [reply]
did you update? 2 months ago is the newest update??

**Reply by kwackers** [reply]
Yep, that's the one.
I fixed it for myself a few months ago but only pushed the fix up when you mentioned it a few days ago.

**Reply by tomvikse** [reply]
it is this that have the bug.. downloaded and tested it 7
 days ago.. testing it out now

**Reply by kwackers** [reply]
I only pushed up the fix 3 days ago... ;)

**Reply by tomvikse** [reply]
ok

**Reply by tomvikse** [reply]
Working good now..thanks :-).. only thing is retract dont stop the same place every time.. differ 1-2mm,not a big deal.. this is runnig stock speed on leed skrew.

**Reply by kwackers** [reply]
Yep, retract can't stop at exactly the same place.
It needs to allow for the position of the head so it can move back and forth by a the pitch depending on where the headstock is when you retract.

**Comment by R-Ronald** [comment]
Not sure this is a good place to get some help on a Thread to Shoulder problem, but I thought I'd give it a try.

I got my ELS installed and running around January of 2024. Switched over to this branch shortly thereafter. Since then, I've been using the lathe for cutting, but not threading. ELS has worked great.  Today was the first time I got around to cutting a thread. Unfortunately, something is wrong.

I appears that "cutting" part of "thread to shoulder" is working properly. It's doing the right thread pitch and it looks good after the first pass. Note the scratch pass in the blued area.
![firstpass](https://github.com/kwackers/electronic-leadscrew/assets/31015839/a44185ae-7781-485e-ab0f-50217b5dcc17)
But after the retract and return to the start location, the 2nd pass does not follow in the footsteps of the first. Somehow, the sync for the start appears to have been lost. The half-nut lever was not touched:
![2ndpass](https://github.com/kwackers/electronic-leadscrew/assets/31015839/353af470-59de-4c89-b6da-1f66ad14baf5).

The smaller diameter section protruding further from the collet (without any blue) shows what happens if one is hardheaded and continues to do many passes. It appears that the start point is essentially random. 

In this case of the blued diameter, the pitch is 1.5mm and it appears that the 2nd pass is off by something in the ballpark of .5mm.

Any ideas on what I should be looking at?

RR

**Comment by LabRatJason** [comment]
Note: I didn’t write the kwackers mod, but I am successfully using itMy first place to look would be with ratios:  is your spindle to encoder ratio 1:1?  Also, what encoder are you using?  Are you using micro steps to adjust your leadscrew ratio?  The reason I ask, is that the math to get you back to zero requires a few assumptions (I know, I tried to code my own stop and resync, and gave up when I discovered this version). One of those may be the source of the resync issue.  I know when James did the video walkthrough on setting up your parameters he shows using the micro steps parameter to adjust the ratio of the spindle, encoder.  Mine is set 1:1 so I can’t have a problem there, but if you are using a weird ratio there it’s likely a problem. The other likely culprit could be if you aren’t using a 1024 step encoder it could be a wrap-around issue. Sent from my iPhoneOn Apr 4, 2024, at 11:37 AM, R-Ronald ***@***.***> wrote:﻿
Not sure this is a good place to get some help on a Thread to Shoulder problem, but I thought I'd give it a try.
I got my ELS installed and running around January of 2024. Switched over to this branch shortly thereafter. Since then, I've been using the lathe for cutting, but not threading. ELS has worked great.  Today was the first time I got around to cutting a thread. Unfortunately, something is wrong.
I appears that "cutting" part of "thread to shoulder" is working properly. It's doing the right thread pitch and it looks good after the first pass. Note the scratch pass in the blued area.
firstpass.jpg (view on web)
But after the retract and return to the start location, the 2nd pass does not follow in the footsteps of the first. Somehow, the sync for the start appears to have been lost. The half-nut lever was not touched:
2ndpass.jpg (view on web).
The smaller diameter section protruding further from the collet (without any blue) shows what happens if one is hardheaded and continues to do many passes. It appears that the start point is essentially random. in this case, the pitch is 1.5mm and it appears that the 2nd pass is off by something in the ballpark of .5mm.
Any ideas on what I should be looking at?
RR

**Reply by R-Ronald** [reply]
Here are some parameters from the code I am running. I tried to use the same hardware and paramaters (as much as possible) from Clough42's site.
//Encoder (**I use the recommended encoder**)
#define ENCODER_RESOLUTION 4000
// Steps and microsteps (**I use the iSV57T-180S that seems to be the favorite. I'm using the 2:1 gearbox reduction to get more torque out of the servo**)
#define STEPPER_MICROSTEPS 3
#define STEPPER_RESOLUTION 1000
// Separate step and microstep settings for feed rates.  **(I determined the feed rate experimentally since I have a separate feed bar. This doesn't impact movement using the lead screw, so it should not be a factor**)
#define STEPPER_MICROSTEPS_FEED 4.71
#define STEPPER_RESOLUTION_FEED STEPPER_RESOLUTION

#define LEADSCREW_TPI 8

#define backlash 2

#define leadscrewRPM(rpm) ((200000 * 60) / rpm) / (STEPPER_MICROSTEPS * STEPPER_RESOLUTION);
// default to 200rpm (**I haven't changed this, but it appears the comment is wrong and the retract is only 100 RPM**)
const int32 retractSpeed = leadscrewRPM( 100 );

#define USE_FLOATING_POINT

#define MAX_BUFFERED_STEPS 100

**None of the CPU/TIMING values have been modified**

**Comment by LabRatJason** [comment]
If you are using the Omrom encoder that James suggested, it’s 1024 steps with quadrature encoding, so it should be 4096 rather than 4000 for the encoder resolution setting. Changing that will probably cause all kinds of havoc with your other parameters so it may be good to look over the spec sheets of the other equipment you are running just to verify the settings are correct. If you are truly using an encoder with 1000 steps (and therefore 4000 with quadrature) then your issue lies somewhere else. -Jason  Sent from my iPhoneOn Apr 4, 2024, at 5:55 PM, R-Ronald ***@***.***> wrote:﻿
Here are some parameters from the code I am running. I tried to use the same hardware and paramaters (as much as possible) from Clough42's site.
//Encoder (I use the recommended encoder)
#define ENCODER_RESOLUTION 4000
// Steps and microsteps (I use the iSV57T-180S that seems to be the favorite. I'm using the 2:1 gearbox reduction to get more torque out of the servo)
#define STEPPER_MICROSTEPS 3
#define STEPPER_RESOLUTION 1000
// Separate step and microstep settings for feed rates.  (I determined the feed rate experimentally since I have a separate feed bar. This doesn't impact movement using the lead screw, so it should not be a factor)
#define STEPPER_MICROSTEPS_FEED 4.71
#define STEPPER_RESOLUTION_FEED STEPPER_RESOLUTION
#define LEADSCREW_TPI 8
#define backlash 2
#define leadscrewRPM(rpm) ((200000 * 60) / rpm) / (STEPPER_MICROSTEPS * STEPPER_RESOLUTION);
// default to 200rpm (I haven't changed this, but it appears the comment is wrong and the retract is only 100 RPM)
const int32 retractSpeed = leadscrewRPM( 100 );
#define USE_FLOATING_POINT
#define MAX_BUFFERED_STEPS 100
None of the CPU/TIMING values have been modified

**Comment by kwackers** [comment]
Thread to shoulder relies on being able to multiply the number of threads by the pulses per rev to return to the correct place.
For the start position to be random the number of pulses per rev must be wrong.

In operation when it stops at the shoulder it keeps track of where it is by counting pulses. 
So for example if you have 2000 pulses per rev but you've set 2048 then each revolution it's not moving the start position will drift by 48 pulses and when it returns this will cause the apparent random start position your seeing.

Hope this helps,

**Comment by R-Ronald** [comment]
My encoder order was for: **E6B2-CWZ6C For OMRON Rotary Encoder Rotary switch 1024P/R 6000r/min 5-24V USA**

So, I'm officially a knucklehead :-)

However, I'm quite happy to be proved wrong since it means that I have an obvious fix. Far better than dragging an oscilloscope out to the shop. 

This seems to match the observed drift quite well. I was cutting between 8 and 9 threads in the two pictures. With 4000 vs 4096, I was 96 short per rev. So roughly 800-900 pulses short across the entire cut. About 20 to 25% of a revolution.

I was planning on measuring how far off I was with the DRO this morning for the shoulder and start positions as a troubleshooting step. It might have hinted at this problem. But, I can skip that now and reload the firmware with the correct P/R number.

I'll report back with results.

**Comment by R-Ronald** [comment]
Success!
![FixedThreading](https://github.com/kwackers/electronic-leadscrew/assets/31015839/70177e9c-7146-49e9-acf6-e72af4a70456)
It's amazing how precise the control is. The stopping location of the carriage for the shoulder seems to consistently be within 2 or 3 microns.
I definitely owe the two of you a beer. Thanks so much for the help!

**Reply by R-Ronald** [reply]
I need to ask for help again. Frustratingly, my problem has re-occurred. 

The shoulder position is always hit with amazing precision. The DRO shows the shoulder position (that I marked as zero on the DRO) always coming back to 0.0000 or -0.0004". Perfect.

But, the start position varies. I measured it on one threading attempt. When I set START, the DRO read 2.0406". And the spindle was marked at the top(0 degrees) for angular position. 

- 2nd pass was 2.0950". The spindle was ~225 degrees
- 3rd pass was 2.0903". The spindle was ~85 degrees
- 4th pass was 2.0921". The spindle was ~190 degrees.
- 5th pass was 2.0752". The spindle was ~5 degrees.

So, the start varies quite a bit. The 2nd pass was 54 thousandths of from the first. Probably a coincidence, but I was cutting an 18 TPI thread and the distance for each thread is .055". The other starts are closer, but the 5th one is about 20 thousandths away from the 2nd one.

My first thought was that I'd somehow loaded the wrong constant for the encoder again. I brought the laptop out and reloaded the code after verifying that the encoder constant was set to 4096 P/R. The problem did not change.

Any other ideas on what might be causing this? Is the start position variance the problem?

**Reply by tomvikse** [reply]
[kwackers](https://github.com/kwackers)
[on Dec 14, 2022](https://github.com/kwackers/electronic-leadscrew/discussions/3#discussioncomment-4401813)
Yep, retract can't stop at exactly the same place.
It needs to allow for the position of the head so it can move back and forth by a the pitch depending on where the headstock is when you retract.

**Comment by kwackers** [comment]
Just to elaborate.
The cutter has to move to the position it would be on the thread for the current head position.

For example if your head stopped at a notional zero degrees and the head stopped at X.
Then the next time if your head was stopped at 180 degrees the cutter can't stop at X, it has to stop half the pitch in front or behind X.

You could rotate the head to a known position before returning to start - in theory it would always stop at the same position or better just make sure there's a little space at the start for the cutter.

**Reply by R-Ronald** [reply]
I did run a separate test where I rotated the head back to zero each time. Did not seem to help. I'll try it again and record another set of resulting START positions and spindle angle (which should be zero each time).

I'm also going to try cutting a thread w/o using the thread to shoulder function. I'll just turn off the lathe at the end of the cut, then  retract, and reverse the lathe with the tool retracted to bring it back it back to (or beyond) the start. I'll need big run-off spaces in both directions since the spindle will keep turning and the leadscrew will keep moving the carriage due to inertia. This should make sure there's no loss of sync from cutting and reversals. If that works properly, that should essentially narrow the problem down to the retract function.

I also inspected my encoder along with the gears and belt driving it. I didn't notice anything wrong. The belt is tight enough that it can't slip, but not too tight. If it wasn't consistent, I would expect the thread to wander during a single cut, which it doesn't.

**Reply by kwackers** [reply]
I'm not sure what the issue is.
Is the cutter not synchronised with the thread?
Is it not stopping at the shoulder?

Unless there's an issue with either of those then it's working as it should.

If it's both synchronised and stopping at the right place then it's always worked like that - retract can't guarantee it'll finish in the correct place even if you think you've stopped at zero you could be a whole pitch out which for your 18tpi thread is a max of 55 thou and your measurements seem to be within that.

Whilst it's possible to stop the retract at exactly the right point the issue then would be the need to "catch up" when you start the lathe. 
To a large extent the system works because the lathe head has to accelerate to speed and with it the ELS can also accelerate to speed.
But if you'd stopped a whole revolution away then when you start the lathe the headstock would have a whole revolution to speed up and then the ELS would have to accelerate the motor way past it's physical capabilities and the result would be a stall and loss of sync.

As I said though this is how the system has always worked, potentially it can always go past the start position by a single thread.

**Reply by R-Ronald** [reply]
The problem is that the cutter is not always synchronized with the thread. It stops "perfectly" at the shoulder.

**Comment by LabRatJason** [comment]
Chipping in to mention this:During retract your leadscrew is turning back a set number of steps, but your DRO is measuring real position.  This means backlash may also be a component of your variance. To test, run it back to start, and turn the chuckForward a full rotation back to “zero degrees” and only then take the value from the DRO. If backlash is occurring I suspect this measure will be much more accurate than what you were seeing before. -Jason Sent from my iPhoneOn Oct 18, 2024, at 1:49 PM, Steve Ward ***@***.***> wrote:﻿
I'm not sure what the issue is.
Is the cutter not synchronised with the thread?
Is it not stopping at the shoulder?
Unless there's an issue with either of those then it's working as it should.
If it's both synchronised and stopping at the right place then it's always worked like that - retract can't guarantee it'll finish in the correct place even if you think you've stopped at zero you could be a whole pitch out which for your 18tpi thread is a max of 55 thou and your measurements seem to be within that.
Whilst it's possible to stop the retract at exactly the right point the issue then would be the need to "catch up" when you start the lathe.
To a large extent the system works because the lathe head has to accelerate to speed and with it the ELS can also accelerate to speed.
But if you'd stopped a whole revolution away then when you start the lathe the headstock would have a whole revolution to speed up and then the ELS would have to accelerate the motor way past it's physical capabilities and the result would be a stall and loss of sync.
As I said though this is how the system has always worked, potentially it can always go past the start position by a single thread.

**Reply by R-Ronald** [reply]
I started to collect some numbers, but then decided to drop back and cut a thread w/o using Thread_to_Shoulder. 

I cut the OD to size, then started taking passes with the thread tool. I was doing an M10 (1.5mm pitch) thread. The half nuts stayed engaged the entire time. I cut towards the chuck and started perhaps 12 mm to the right of the part. I had a "cut area" of about 30mm and had another 10 mm of undercut for runoff. Cut left, turn off spindle, retract tool, go right (in the air) and turn it off well past the start. Wind in and repeat.

This also failed. It appears that the direct relationship between spindle and lead screw is not being held by my ELS. Looks to me that the Thread-To_Shoulder code is innocent. If I understand, this must either be a problem with the encoder or the servo. 

I did note a "ton" of backlash in the system driving the carriage. I can rotate the spindle about 1/3 of a turn (120 degrees) before the carriage moves in the opposite direction. About 1/2 of this is in the lathe gearbox and about 1/2 is in the carriage. The servo doesn't appear to be a big contributor. I imagine that backlash of some amount is pretty much inherent.

**Comment by LabRatJason** [comment]
That's interesting.  If the cutter is not always synchronized with the
threads, and you have not disengaged the half nuts, then you have either:
a) a mechanical issue (varying torque/pressure causing something mechanical
to slip or bind) or b) you are losing counts on the encoder or steps in the
stepper.  To diagnose, we may need a breakdown of what hardware you have.
Some questions that may help us:
1) how are you driving the encoder?  If it is by belt, is it a toothed belt?
2) Is the encoder electrically isolated?  It should ONLY be connected to
the circuit board, it should not be grounded to the chassis.
3) How are you powering your stepper motor?  What voltage?  What size
stepper are you using?
4) How is the stepper connected to your leadscrew?  Again, if it is a belt,
is it toothed?

Finally, I will caveat this by saying, I'm still leaning towards
backlash/mechanical issues, because otherwise, how would the cutter stop at
precisely the correct position every time?  That one fact alone casts doubt
on all the troubleshooting questions above.
Perhaps, rather than relying on your DRO, you could verify with a magnetic
base indicator that the carriage is stopping at the correct position?  I
just can't shake the feeling it's a mechanical thing.

-J

**Reply by R-Ronald** [reply]
1) Toothed belt for the encoder.
2) I'm using non-conductive plastic parts for the encoder mounting. It should be well isolated (see pic).
3) Servo is a NEMA 23  Integrated Easy Servo Motor 180w 3000rpm (iSV57T-180S) from stepperonline.com powered by a MEAN WELL LRS-350-48 350.4 Watt supply. 
4) Toothed belt for the servo too.

I agree with your suspicion about mechanical issues. If I touch the bolt for the servo to the gears, it feels like the motion is less than direct. If I drag the gear on the servo, it feels like there is some sort of cogging (even though there's no way the output belt is slipping). Hard to explain, but I don't get the impression that the link is absolutely direct. But the cutter stooping exactly on its mark pretty much requires that to be working.

I'll can check the shoulder position with a dial indicator tomorrow.

RR

![ELS](https://github.com/user-attachments/assets/7f535f9b-b2c5-4945-ac08-324ee5470812)

**Reply by R-Ronald** [reply]
I thought I remembered turning the supply voltage down a bit and thought it would be worthwhile to measure it. So I set my meter to DC volts and touched the probes. But, I neglected to notice that the red meter cable was plugged into the 10A current socket and not the one for voltage. Doh! Big spark....and now no power.

It looks like Meanwell supplies have a fuse. If I'm lucky I'll just have lost one in the meter and one in the power supply.

If I'm unlucky and need a new supply....it will be a tough call. I've been thinking about upgrading to the 400 watt servo motor which needs a bigger supply.

**Reply by kwackers** [reply]
It's been a while since I looked at the code, but the stopping at the shoulder is likely a function of where it thinks the stepper position is.
Any loss of position on the thread is likely to be due to an encoder issue.
If I read you correctly you're saying it's not cutting threads accurately regardless of the mode you're in but is stopping at the correct position for the shoulder - which screams "encoder" to me. If it was a stepper/driver issue it wouldn't stop at the shoulder.

If like me you're using one of the cheap Chinese encoders then I have wondered about their longevity since the "genuine" versions are quite 'spendy' - or you could just be unlucky.

**Reply by R-Ronald** [reply]
Spendy understates things. If I look at Mouser (a distributor in the USA), their OMRON listings seem to range from about $400 to about $800 each. But they don't even have the exact model #. Seems insanely expensive for it. 

 Versus a typical $25 to $50 on Amazon . And all over the place at eBay. But the eBay ones do seem to attract reviews indicting early failure or unreliable operation.

I'm going to order one on Amazon. It's a brand called "Jeanoko". At least they aren't claiming to be Omron. And all three reviews at Amazon indicated that the unit is installed into a Clough42 ELS and are working. All three also mention that the purchaser got bad ones on eBay that did not work. It's $48, but seems worth a try.

**Comment by R-Ronald** [comment]
ELS is back up and running. Only took 65 minutes to remove and check supply and remove and check servo. And reinstall all. No problem found. When I saw the correct voltage and retested the entire ELS sub-system, it worked.

Supply was at 41.3 Volts. I turned it up to 44.3. The Integrated Servo notes indicate that a higher voltage is only useful for increasing the speed of the servo, but not the torque. But, it's rated to a max of 50 volts (even though the nominal power supply is 36 Volts). I don't think it can hurt. I tightened the servo belt a tiny bit.

New encoder is supposed to arrive Monday. I may wait till then for further testing. As listed above, I have seen the encoder causing the same apparent problem when I had entered the wrong constant for the P/R value. So, I'm sure an erratic sensor could create the same problem. And there are apparently MANY people who used the direct eBay link off Clough42's ELS Bill of Materials that have had problems with erratic or completely failed sensors. And that's where I sourced the encoder too. So it seems a good bet. 

One more thing that makes this a good hypothesis. I did cut a thread last week that looks pretty good. So, it is apparently a somewhat intermittent problem. Which makes hardware much more likely than software.

**Reply by R-Ronald** [reply]
Received the new encoder today. Installed it, which took much more time than it should. 

Looks good. I did more than 20 passes and did not see any variance from the proper path. I didn't start with a correct OD and didn't spend time to converge on the proper PD for any particular thread...but I'm satisfied that the old encoder must have been the problem.
![goodThread](https://github.com/user-attachments/assets/2e984a19-c909-436a-9af4-d2be68ffb5a8)

I recorded the start position for the first 17 passes along with the spindle angle. For the last 4 passes, I roughly aligned the spindle to an identical position (perhaps + or - 5 degrees). For those 4 passes, the start was 43.33mm, 43.30, 43.289, and 43.287. So all four starts were within .043 mm.....about 1.6 thousandths of an inch.

Going forward, this should be a trivially easy problem to diagnose. And anyone building the ELS (with the thread to should code) probably ought to run this test before declaring victory. Unless they can afford a real Omron encoder :-)

You might want something like the text below in the FAQ. 

Using thread to shoulder to verify the servo and the encoder. You won't actually cut anything except air during this test:

1. Set the ELS to metric or inch-based as desired.
2. Select the desired thread pitch.
3. Use the set button and then the up arrow button to display "thread to shoulder"
4. Hit the set button to select thread to shoulder.
5. Hit the set button again to select single thread start
6.  Hit the set button again to set the start location. Put the lathe tool where you want the thread to end. Engage the half nuts. Make sure you have threading enable in the gearbox. Hit "set" again.
7. Make sure the speed is appropriate for the test. There is no minimum speed required since you aren't actually cutting any metal. In fact, you don't even need to mount a cutter for the test. You just want the lathe slow enough that you don't crash into anything.
8.  Move the carriage to the start position for the thread by hand rotating the spindle or by using reverse on the lathe. Give the lathe a bit of room for ramping up speed before the actual starting point you want. Hit "set" again and you should see "Begin" in the display.
9. Start the lathe and it should move from the start position to the shoulder position whereupon the carriage should stop moving. The spindle will still be rotating.
10. Turn off the spindle.
11. At this point, you will see "retract" on the display. If you were actually cutting something, you'd need to crank the cutter out at this point. But, if you're just testing (recommended), it's not near any metal and no retraction is needed. 
12. Before doing anything else, record the Z position at the shoulder point. Use a DRO or a dial indicator. All subsequent stops at the shoulder should be very close to the same place. On my machine, they are typically within 10 microns. If your readings vary significantly, you may have a servo problem.
13. After recording the start position, take a sharpy or use a couple of pieces of tape to mark the spindle position. I made a mark on the chuck and a mark on the body of the lathe. Each time you come back to the shoulder, you will need to adjust the rotation of the spindle (rotating it by hand) so that it matches the angular position of this pass. By doing so, you will make the distance that the ELS needs to move the carriage  identical and measurable. If the angular position of the spindle varies, the ELS will need to change the start location to compensate. By making it the same, you eliminate the need for compensation and you make it possible to verify that the start position is consistent.
14. Hit the "up arrow" button and the carriage will move from the shoulder to the start location. Measure the start location using a DRO or a dial indicator. Record the position.
15. You should have "begin" on the ELS display. If you were cutting, you'd advance the cutter each time you reach this point. But, if you're just testing there's nothing to do here now except to start the lathe and make the 2nd pass.
16. When the pass completes, verify it's in the same place as the last time around. If it is, your servo is doing its job.
17. Align the marks for the spindle angle by hand-rotating the spindle.
18. Hit the "up arrow" button.
19. When the carriage gets back to the start position, record it. 
20. Compare the 1st start position to the 2nd. They should be very close. Any error in aligning the spindle to the same angular position will result in a change in the start position. That's because the ELS trades angular position for distance in order make the thread helix start at exactly the same place in both distance and angle. If you are cutting 20 TPI, the spindle revolves once for every .05" of carriage movement. So, if the spindle is 180 degrees out of phase with a previous pass, the start distance will be offset exactly 1/2 the .05".
21. Continue making passes and recording start and stop positions until you are happy with the results OR you see enough evidence to indicate a problem. If the start position is not consistent, the likely suspect is the spindle angular position encoder.

The eBay vendor where Clough42 sourced his "Omron" encoder is suspect. There are many reports of failures, some of which are intermittent and occasional. Genuine Omron encoders are sold for $400-$800, so it seems likely that most of the eBay and Amazon "Omron" encoders are counterfeits. **Buyer beware!** It's probably better to purchase from a knock-off vendor that doesn't deceive buyers about the company making the product.

**Comment by almar50** [comment]
Does anybody have any idea what could cause  the same problem as tomvikse in the first post? Everything seems to work fine except the thread to shoulder that once it reaches the shoulder, there is no "stop". When I stop the spindle, it just says. SH 0...not "stop".... The weird thing is that sometimes it does work, maybe once every 4 or 5 times. It always stops right at the shoulder. But there is no continuance of the program after on most cycles.

Edit: all the pulleys and belts seem tight.
 Everything seems to be plugged in correctly. And the thread seemed to track well with previous passes If I continue manually. I'm just curious as to what could be the culprit that prevents it from initiating the stop sequence. The servo knows to stop so the encoder counted the pluses correctly...but the code breaks and dosent continue? I tried 2 different launch pads and encoders...same thing. It can't be the code because it works for others.

**Comment by almar50** [comment]
Well i fixed that. It wouldn't work with a stepper resolution of a 1000 but works with 200. Unfortunately, my threads are misaligned due to the retract.

**Reply by kwackers** [reply]
Maths overflow?

**Reply by almar50** [reply]
I looked at the code and thats what i think the problem was which is why i backed it down to 200. Or at least hoped it was. Now i have to figure out the missalignment problem...the normal threading is prefectly aligned so that narrows it down somewhat...

**Reply by almar50** [reply]
So I have an encoder problem. I am using the good ones. The one from amazon and also the more expensive japanese one. If I mark the position of the spindle and select position to give me an angle and start out at 0 deg, i lose about five degrees for every rotation, and they add up. I find that suspect that both would give me the same problem though... and that I don't have any issues with misalignment. When using the standard clough42 threading program...smh...

**Comment by LabRatJason** [comment]
Could this be a mismatch between 1000 step and 1024 step encoders?-JasonSent from my iPhoneOn Sep 11, 2026, at 10:22 AM, almar50 ***@***.***> wrote:﻿
So I have an encoder problem. I am using the good ones. The one from amazon and also the japanese more expensive one. If I mark the position of the spindle and select position to give me an angle and start out at 0 deg, i lose about five degrees for every rotation, and they add up.

**Reply by almar50** [reply]
But my input in the configuration h file is for a 1024pr (4096) and both my encoders are 1024...

**Comment by LabRatJason** [comment]
What does your fear train look like from spindle to encoder?JasonSent from my iPhoneOn Sep 11, 2026, at 10:42 AM, almar50 ***@***.***> wrote:﻿
But my input in the configuration h file is for a 1024pr (4096) and both my encoders are 1024...

**Reply by almar50** [reply]
Its a 1:1 gt2 pulley system. There's a little bit of run out in the spindle pulley...but the encoder is on a spring loaded bracket and the teeth dont slip.. The angular position drops at an inconsistent rate between two degrees and six degrees for every rotation. Eight full rotations dropped from 360 deg(or zero degrees), to 326.8 degrees.

**Reply by almar50** [reply]
I marked the pulleys and the spindle. With fixed reference points at zero degrees after several rotations, they all line up perfectly.

**Reply by almar50** [reply]
The drop is inconsistent for every rotation, but it's consistently dropping. I'm wondering if I can average the drops out and factor down the encoder resolution accordingly to compensate for the error.

**Comment by LabRatJason** [comment]
My next guess would be lost steps in the encoder due to electrical noise?  I’m not sure how the encoder signal works, it’s been a long time since I’ve thought about mine, but I’d be very hesitant to try to “feel it out” with making adjustments.  Your threads will never work out right and you’ll risk crashes when threading up to a shoulder. Check for loose connections and signal wires too close to high frequency sources. JasonSent from my iPhoneOn Sep 11, 2026, at 11:46 AM, almar50 ***@***.***> wrote:﻿
The drop is inconsistent for every rotation, but it's a consistent drop. I'm wondering if I can average them out and factor down the encoder resolution accordingly to compensate for the error.

**Reply by almar50** [reply]
Yes, noise is a big concern. I've been working on that since early this morning. But nothing I found so far helped. The last thing I have to try is my enclosure. Fan, that I put a bit too close to the launch pad. By the way, I appreciate the help. I know it can be difficult to do that remotely.

**Comment by almar50** [comment]
This is very odd, but I did the adjustment to the encoder resolution, as I mentioned before, and it showed an increase in angle instead of a reduction. The increase was exactly the amount I adjusted it by.... I reflashed, it again at 4096 and did another test and now it's exactly 357.3 degrees at the mark. After several rotations.... This is random stuff...

**Comment by LabRatJason** [comment]
I’ve actually never done this test on my lathe, because I haven’t had a need to do angles. But I can run out and try this tonight to see if perhaps this is more an issue of just the angle measurements rather than the encoder chuck relationship.JasonSent from my iPhoneOn Sep 11, 2026, at 12:45 PM, almar50 ***@***.***> wrote:﻿
This is very odd, but I did. The adjustment to the encoder resolution, as I mentioned before, and it showed an increase in angle instead of a reduction. The increase was exactly the amount I adjusted it by.... I reflashed, it again at 4096 and did another test and now it's exactly 357.3 degrees at the mark. After several rotations.... This is random stuff...

**Comment by almar50** [comment]
Maybe it is the angle measurement.... But I'm pretty much out of ideas at this point. For the misalignment.... Maybe it is bad encoders... Buying an omron will be pricey and I don't even know if that's the problem.

**Comment by LabRatJason** [comment]
Unfortunately, when I tasted my lathe, which is on 1.6.0, my angle measurements were quite accurate. I put a construction laser on the tailstock and two round magnets on the chuck, so that the laser passed perfectly on the edge between the two magnets (side by side and touching). I ran it for about a minute at 500 rpm and checked again and it was dead on. Not sure where to tell you to look next. I’m using a genuine omrom encoder, and like you I run a 1:1 ratio with the chuck (technically there is a 1:2 gear coming off the chuck, so I use a 2:1 to undo that.)JasonSent from my iPhoneOn Sep 11, 2026, at 3:44 PM, almar50 ***@***.***> wrote:﻿
Maybe it is the angle measurement.... But I'm pretty much out of ideas at this point. For the misalignment.... Maybe it is bad encoders... Buying an omron will be pricey and I don't even know if that's the problem.

**Comment by almar50** [comment]
Ok...so ill work on this and try to fix it. There are a few more things that i can try that i just thought of.

**Reply by almar50** [reply]
In the hope that one day, this becomes useful to somebody else. I think I found the problem, or at least I found a significant contributor to it. The encoder belt to spindle was slightly twisted, barely perceptible, but it was there one side of the belt was lifting higher than the other. I don't think I would have ever considered this to be an issue even if I had seen it, but it was, because when I removed it, the angular error is removed. Even after running the lathe at high r p m. I'm looking into possibly replacing the belt mechanism with gears... The only reason I discovered this is because, during the testing the error kept changing every time I bumped the belt with my hand which led me to looking closer at it.

**Comment by LabRatJason** [comment]
Yeah, you should be using a toothed belt or gears. Anything else will slip. JasonSent from my iPhoneOn Sep 12, 2026, at 9:28 AM, almar50 ***@***.***> wrote:﻿
In the hope that one day, this becomes useful to somebody else. I think I found the problem, or at least I found a significant contributor to it. The encoder belt to spindle was slightly twisted, barely perceptible, but it was there one side of the belt was lifting higher than the other. I don't think I would have ever considered this to be an issue even if I had seen it, but it was, because when I removed it, the angular error is removed. Even after running the lathe at high r p m. I'm looking into possibly replacing the belt mechanism with gears...

**Reply by almar50** [reply]
Yeah agreed, I am using a gt2 belt which is toothed but it wasn't tracking perfectly.
